mojo-executor-maven-plugin/src/main/java/org/twdata/maven/mojoexecutor/plugin/MojoExecutorMojo.java

```diff
@@ -83,13 +89,47 @@
      */
     private BuildPluginManager pluginManager;
 
-    public void execute() throws MojoExecutionException {
-        if ( mavenProject.getArtifactId().equals( "mojo-executor-test-project-null-maven-project" ))
+    public void execute() throws MojoExecutionException
+    {
+        integrationTestSetup();
+
+        getLog().info( "Executing with maven project " + mavenProject + " for session " + mavenSession );
+        executeMojo( plugin, goal, toXpp3Dom( configuration ),
+                     executionEnvironment( mavenProject, mavenSession, pluginManager ) );
+    }
+
+    /**
+     * Used purely to customise the setups for the integration tests.
+     *
+     * @throws MojoExecutionException
+     */
+    private void integrationTestSetup() throws MojoExecutionException
+    {
+        // Test specific conditionals...
```

Comment:
Thats an interesting idea :-) Another idea is we make the plugin 'configurable' which would make the example more interesting and break the two way link ; the tests would activate whichever configuration they require....